mautrix_telegram/__main__.py

```diff
@@ -271,6 +284,54 @@ async def manhole_global_namespace(self, user_id: UserID) -> dict[str, Any]:
             "Puppet": Puppet,
         }
 
+    async def _notify_bridge_blocked(self, is_blocked: bool = True) -> None:
+        msg = self.config["bridge.limits.block_ends_notification"]
+        if is_blocked:
+            msg = self.config["bridge.limits.block_begins_notification"]
+            next_notification = self._last_blocking_notification + self.config["bridge.limits.block_notification_interval_seconds"]
+            # We're only checking if the block is active, since the block-end notification will not be resent and we want it ASAP
+            if next_notification > int(time()):
+                return
+            self._last_blocking_notification = int(time())
+
+        admins = list(map(lambda entry: entry[0],
+            filter(lambda entry: entry[1] == 'admin',
+                self.config["bridge.permissions"].items()
+            )
+        ))
+        if len(admins) == 0:
+            self.log.debug('No bridge admins to notify about the bridge being blocked')
+            return
+
+        self.log.debug(f'Notifying bridge admins ({",".join(admins)}) about bridge being blocked')
+
+        if not self._admin_rooms:
+            self.log.debug("Fetching admin rooms from the homeserver")
+            admin_rooms = {}
+            joined_rooms = await self.az.intent.get_joined_rooms()
+            for room_id in joined_rooms:
+                members = await self.az.intent.get_room_members(room_id)
+                if len(members) == 2: # a DM with someone
+                    for admin_mxid in admins:
+                        if admin_mxid in members:
+                            admin_rooms[admin_mxid] = room_id
+                            break
+            self._admin_rooms = admin_rooms
+
+        for admin_mxid in admins:
+            if admin_mxid not in self._admin_rooms:
+                self.log.debug(f"Creating a new admin room for {admin_mxid}")
+                self._admin_rooms[admin_mxid] = await self.az.intent.create_room(
+                    name="Telegram Bridge alerts",
+                    invitees=[admin_mxid],
```

Comment:
I don't know how this library handles it, but in JS / the Matrix API, this room creation fails if the invitee is invalid or their homeserver doesn't answer as expected for whatever reason (e.g. network outage, maintenance, busy, disk fill).

Sending a separate invite call to the API reduces the risk of a room creation failure. Then again, we probably don't want that room, if the invite fails.